Kernel/Graphics/GraphicsManagement.cpp

```diff
@@ -56,7 +56,74 @@ void GraphicsManagement::activate_graphical_mode()
 
 static inline bool is_vga_compatible_pci_device(PCI::Address address)
 {
-    return PCI::get_class(address) == 0x3 && PCI::get_subclass(address) == 0x0;
+    // Note: Check for Display Controller, VGA Compatible Controller or
+    // Unclassified, VGA-Compatible Unclassified Device
+    return (PCI::get_class(address) == 0x3 && PCI::get_subclass(address) == 0x0) || (PCI::get_class(address) == 0x0 && PCI::get_subclass(address) == 0x1);
+}
+
+static inline bool is_display_controller_pci_device(PCI::Address address)
+{
+    return PCI::get_class(address) == 0x3;
+}
+
+UNMAP_AFTER_INIT bool GraphicsManagement::determine_and_initialize_graphics_device(const PCI::Address& address, PCI::ID id)
+{
+    VERIFY(is_vga_compatible_pci_device(address) || is_display_controller_pci_device(address));
+    auto add_and_configure_adapter = [&](GraphicsDevice& graphics_device) {
+        m_graphics_devices.append(graphics_device);
+        if (!m_framebuffer_devices_allowed) {
+            graphics_device.enable_consoles();
+            return;
+        }
+        graphics_device.initialize_framebuffer_devices();
+    };
+
+    RefPtr<GraphicsDevice> adapter;
+    switch (id.vendor_id) {
+    case 0x1234:
```

Comment:
The amount of bare numbers without any sort of comment in the kernel (especially in the driver side) is somewhat upsetting to me.
Specifically in this case, more so since we already have these in `Bus/PCI/IDs.h`, as demonstrated by `VirtIO` below.

Reply:
You're always free to invest time in changing these bare numbers in other parts of the kernel :)
I'd so in a separate commit, it doesn't belong to the first commit.

Reply:
certainly, but why not do so for the values _that already exist_?
Anyhow, if you're not interested in dealing with them, I'll mark my comments about them as resolved.